code/modules/antagonists/fugitive/hunter.dm

```diff
@@ -0,0 +1,171 @@
+			to_chat(owner, "<B>The criminals should be on the station, we have special huds implanted to recognize them.</B>")
+			to_chat(owner, "<B>As we have lost pretty much all power over these damned lawless megacorporations, it's a mystery if their security will cooperate with us.</B>")
+		if("russian")
+			to_chat(src, "<span class='danger'>Ay blyat. I am a space-russian smuggler! We were mid-flight when our cargo was beamed off our ship!</span>")
+			to_chat(src, "<span class='danger'>We were hailed by a man in a green uniform, promising the safe return of our goods in exchange for a favor:</span>")
+			to_chat(src, "<span class='danger'>There is a local station housing fugitives that the man is after, he wants them returned; dead or alive.</span>")
+			to_chat(src, "<span class='danger'>We will not be able to make ends meet without our cargo, so we must do as he says and capture them.</span>")
+
+	to_chat(owner, "<span class='boldannounce'>You are not an antagonist in that you may kill whomever you please, but you can do anything to ensure the capture of the fugitives, even if that means going through the station.</span>")
+	owner.announce_objectives()
+
+/datum/antagonist/fugitive_hunter/create_team(datum/team/fugitive_hunters/new_team)
+	if(!new_team)
+		for(var/datum/antagonist/fugitive_hunter/H in GLOB.antagonists)
+			if(!H.owner)
+				continue
+			if(H.hunter_team)
+				hunter_team = H.hunter_team
+				return
+		hunter_team = new /datum/team/fugitive_hunters
+		hunter_team.update_objectives()
+		return
+	if(!istype(new_team))
+		stack_trace("Wrong team type passed to [type] initialization.")
+	hunter_team = new_team
+
+/datum/antagonist/fugitive_hunter/get_team()
+	return hunter_team
+
+/datum/team/fugitive_hunters
+	var/backstory = "error"
+
+/datum/team/fugitive_hunters/proc/update_objectives(initial = FALSE)
+	objectives = list()
+	var/datum/objective/O = new()
+	O.team = src
+	objectives += O
+
+/datum/team/fugitive_hunters/proc/get_all_fugitives()
+	. = list()
+	for(var/mob/crew in GLOB.mob_list)
+		if(crew.mind)
+			if(crew.mind.has_antag_datum(/datum/antagonist/fugitive))
+				. += crew
+
+/datum/team/fugitive_hunters/proc/fugitives_dead(list/fugitives_counted)
+	. = 0
+	for(var/mob/living/L in fugitives_counted)
+		if(!istype(L))
+			.++
+		if(L.stat != DEAD)
+			.++
+
+/datum/team/fugitive_hunters/proc/fugitives_captured(list/fugitives_counted)
+	. = 0
+	for(var/mob/living/L in fugitives_counted)
+		var/datum/antagonist/fugitive/fug = L.mind.has_antag_datum(/datum/antagonist/fugitive)
+		if(fug.is_captured == TRUE)
+			.++
+
+/datum/team/fugitive_hunters/proc/all_hunters_dead()
+	var/dead_boys = 0
+	for(var/I in members)
+		var/datum/mind/hunter_mind = I
+		if(!(ishuman(hunter_mind.current) || (hunter_mind.current.stat == DEAD)))
+			dead_boys++
+	return dead_boys >= members.len
+
+/datum/team/fugitive_hunters/proc/get_result()
+	var/list/fugitives_counted = get_all_fugitives()
```

Comment:
Also generally if you only use these helpers here it could be just one loop. But i suppose there's some utility here so it's not a big deal.